go/cmd/dolt/commands/diff.go

```diff
@@ -613,7 +619,41 @@ func diffRows(
 
 	defer rowIter.Close(sqlCtx)
 
-	err = writeDiffResults(sqlCtx, sch, unionSch, rowIter, rowWriter)
+	if dArgs.skinny {
+		modifiedColNames, err := getModifiedCols(sqlCtx, rowIter, unionSch, sch)
+		if err != nil {
+			return errhand.BuildDError("Error running diff query:\n%s", query).AddCause(err).Build()
+		}
+
+		// instantiate a new schema that only contains the columns with changes
+		var filteredUnionSch sql.Schema
+		for _, s := range unionSch {
+			for colName := range modifiedColNames {
+				if strings.Contains(s.Name, colName) {
+					filteredUnionSch = append(filteredUnionSch, s)
+				}
+			}
+		}
+
+		// instantiate a new RowWriter with the new schema that only contains the columns with changes
+		rowWriter, err = dw.RowWriter(ctx, td, filteredUnionSch)
+		if err != nil {
+			return errhand.VerboseErrorFromError(err)
+		}
+		defer rowWriter.Close(ctx)
+
+		// reset the row iterator
+		rowIter.Close(sqlCtx)
```

Comment:
(minor) would be good to check any returned `err` here. I think `rowIter.Close` returns an error, but disregard if not.